src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1986,12 +1986,14 @@
               (when-not (or (vec? field)
                             (contains? (set qfields) field)
                             (some #(re-matches % field) (map re-pattern dotted-fields)))
+                (let [[message string] (if (empty? qfields)
+                                         ["Entity {2} has no queryable objects" alias]
+                                         ["Known queryable objects are {2}"
+                                          (utils/comma-separated-keywords qfields)])]
                 {:node node
                  :state (conj state
-                              (format "'%s' is not a queryable object for %s, %s" field alias
-                                      (if (empty? qfields)
-                                        (format "%s has no queryable objects" alias)
-                                        (format "known queryable objects are %s" (json/generate-string qfields)))))}))
+                              (tru (str "''{0}'' is not a queryable object for {1}. " message)
+                                   field alias string))})))
```

Comment:
Does that work?  i.e. assuming the fixed text of the `message` should be translated too, I suspect the translation infrastructure won't be able to see it.  I'm guessing that we'll need the literal strings visible in their entirety in a trs call for each case.

...e.g.

    (if ...
      (tru "''{0}'' is not a queryable object for {1}. Entity {2} has no queryable objects")
      (tru "''{0}'' is not a queryable object for {1}. Known..."))

or something with the same effect.